orchestrator/api/api_v1/endpoints/processes.py

```diff
@@ -370,3 +373,34 @@ def processes_filterable(
         raise CacheHit(HTTPStatus.NOT_MODIFIED, headers=dict(response.headers))
 
     return [asdict(enrich_process(p)) for p in results]
+
+
+@router.websocket("/test/{pid}")
+async def websocket_endpoint(websocket: WebSocket, pid: str, token: str = Query(...)):
+    error = await websocket_manager.authorize(websocket, token)
+
+    await websocket.accept()
+
+    if error:
+        await websocket_manager.disconnect(websocket, reason=error)
+        return
+
+    process = get_current_process_data(pid)
+    await websocket.send_text(json_dumps({"process": process}))
+    if not is_process_running(process):
+        await websocket.close()
+        return
+
+    await websocket_manager.connect(websocket, pid)
+
+
+def is_process_running(p: ProcessTable) -> bool:
+    return (
+        p["status"] == ProcessStatus.RUNNING
+        or p["status"] == ProcessStatus.SUSPENDED
+        or p["status"] == ProcessStatus.WAITING
+    )
```

Comment:
You could also consider something like:

`return p["status"] in [ProcessStatus.RUNNING, ProcessStatus.SUSPENDED, ProcessStatus.WAITING]`